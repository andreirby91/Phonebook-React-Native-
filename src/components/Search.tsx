import React from 'react';
import {View, TextInput, StyleSheet} from 'react-native';

const styles = StyleSheet.create({
  container: {
    padding: 10,
  },
  input: {
    borderWidth: 0.5,
    borderColor: 'grey',
    backgroundColor: '#fafafb',
    paddingHorizontal: 20,
    paddingVertical: 10,
    marginVertical: 10,
    width: 345,
    borderRadius: 20,
    fontSize: 18,
  },
});

const Search = ({
  searchValue,
  onChangeText,
}: {
  searchValue: string;
  onChangeText: any;
}) => {
  return (
    <View style={styles.container}>
      <TextInput
        style={styles.input}
        value={searchValue}
        onChangeText={onChangeText}
        placeholder="Search by names and numbers:"
      />
    </View>
  );
};

export default Search;
