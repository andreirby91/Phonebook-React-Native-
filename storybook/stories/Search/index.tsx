import React from 'react';
import Search from '../../../src/components/Search';

const SearchStory = () => {
  return <Search searchValue="" />;
};

export default SearchStory;
