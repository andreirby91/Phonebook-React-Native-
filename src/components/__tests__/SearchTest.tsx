import React from 'react';
import renderer from 'react-test-renderer';

import Search from '../Search';

it('renders correctly with defaults', () => {
  const button = renderer
    .create(<Search searchValue="" onChangeText={() => {}} />)
    .toJSON();
  expect(button).toMatchSnapshot();
});
