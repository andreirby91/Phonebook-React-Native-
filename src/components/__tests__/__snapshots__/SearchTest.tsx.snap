// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with defaults 1`] = `
<View
  style={
    Object {
      "padding": 10,
    }
  }
>
  <TextInput
    allowFontScaling={true}
    onChangeText={[Function]}
    placeholder="Search by names and numbers:"
    rejectResponderTermination={true}
    style={
      Object {
        "backgroundColor": "#fafafb",
        "borderColor": "grey",
        "borderRadius": 20,
        "borderWidth": 0.5,
        "fontSize": 18,
        "marginVertical": 10,
        "paddingHorizontal": 20,
        "paddingVertical": 10,
        "width": 300,
      }
    }
    underlineColorAndroid="transparent"
    value=""
  />
</View>
`;
